### PR TITLE
refactor: add a method on Info to list files to copy to the package

### PR DESCRIPTION
**Pull request by dzhu**
Since iterating over the files to copy to the package is something that
all package formats need, adding a central place for that logic reduces
duplication.

Also sort the list of files as a step toward making builds more
reproducible.

### CODE REVIEW

File: nfpm.go [hunk 1]
```diff
@@ -230,3 +233,43 @@ func WithDefaults(info *Info) *Info {
 
 	return info
 }
+
+// CopiedFile describes the source and destination of one file to copy into a
+// package and whether it is a config file
```
Comment (caarlos0) [inline nfpm.go:238]:
```suggestion
// package and whether it is a config file.
```

File: nfpm.go [hunk 2]
```diff
@@ -230,3 +233,43 @@ func WithDefaults(info *Info) *Info {
 
 	return info
 }
+
+// CopiedFile describes the source and destination of one file to copy into a
```
Comment (caarlos0) [inline nfpm.go:237]: the name is a bit misleading... its not a copied file, is a file to be copied... 🤔

Comment (dzhu) [reply nfpm.go:237]: Good point. How about "FileToCopy" (and "FilesToCopy" for the method)?

**Comment by caarlos0** [comment]
This pull request has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**